spec/decorators/visit_decorator_spec.rb

```diff
@@ -11,6 +11,33 @@
     end
   end
 
+  describe '#nomis_offender_id' do
+    context 'when the lookup is disabled' do
+      before do
+        switch_off(:nomis_staff_prisoner_check_enabled)
+      end
+
+      it 'returns nil' do
+        expect(subject.nomis_offender_id).to be_nil
+      end
+    end
+
+    context 'when the lookup is enabled' do
+      let(:checker) { double(StaffNomisChecker) }
+      let(:offender) { double(Nomis::Offender, id: 1234) }
+
+      before do
+        switch_on(:nomis_staff_prisoner_check_enabled)
+        mock_service_with(StaffNomisChecker, checker)
+        expect(checker).to receive(:offender).and_return(offender)
+      end
```

Comment:
I  think you can have  a cleaner test setup with only:

```ruby
let(:offender) { Nomis::Offender.new(id:1234)
before do
  switch_on(:nomis_staff_prisoner_check_enabled)
  mock_nomis_with(:lookup_active_offender, offender)
end
```
and remove the `StaffNomisChecker` double and the `expect` in the  `before` block